src/Uno.UI.Runtime.Skia.Android/Microsoft/UI/Xaml/UnoExploreByTouchHelper.cs

```diff
@@ -123,6 +123,8 @@ protected override void GetVisibleVirtualViews(IList<Integer>? virtualViewIds)
 			return;
 		}
 
+		HashSet<DependencyObject> rememberAllVisited = [];
```

Comment:
This can probably be a field. You can `Clear()` it at the end of the method and then reuse the same instance next time

Reply:
Also, can we a a test for this?

Reply:
fixed and added